workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/GameObjectCreatorFromMetadata.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Generic;
+using System.IO;
+using Generated.Improbable;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectCreatorFromMetadata : IEntityGameObjectCreator
+    {
+        private readonly Dictionary<string, GameObject> cachedPrefabs
+            = new Dictionary<string, GameObject>();
+
+        private readonly string workerType;
+
+        public GameObjectCreatorFromMetadata(string workerType)
+        {
+            this.workerType = workerType;
+        }
+
+        public GameObject CreateGameObjectForEntityAdded(SpatialOSEntity entity, WorkerSystem worker)
```

Comment:
Just making sure my old comment doesn't get buried:
Would it make sense to add a method to IEntityGameObjectCreator for something like bool NewlyAddedEntityShouldBeRepresentedByAGameObject? Alternatively, shall we let GetGameObjectForEntityAdded() return a bool? Returning a null GameObject to tell users that an entity doesn't need a GameObject feels like a gotcha

Reply:
that's a good point, let's create a ticket and create a separate pr for that. I really want this in so I can continue testing

Reply:
Created https://improbableio.atlassian.net/browse/UTY-1161